tests/test_simple_target_loop.py

```diff
@@ -0,0 +1,188 @@
+
+import asyncio
+import os
+import unittest
+import time
+import warnings
+import numpy as np
+
+import SALPY_ScriptQueue
+import SALPY_Scheduler
+import lsst.ts.salobj as salobj
+from lsst.ts.scheduler import SchedulerCSC
+
+try:
+    with_scriptqueue = True
+    from lsst.ts import scriptqueue
+except ModuleNotFoundError:
+    with_scriptqueue = False
+    pass
+
+I0 = scriptqueue.script_queue.SCRIPT_INDEX_MULT  # initial Script SAL index
+
+
+@unittest.skipIf(not with_scriptqueue, "Could not import scriptqueue.")
+class SimpleTargetLoopTestCase(unittest.TestCase):
+    """This unit test is designed to test the interaction of the simple target production loop of the Scheduler CSC
+    with the LSST Queue.
+
+    """
+    def setUp(self):
+        salobj.test_utils.set_random_lsst_dds_domain()
+        self.datadir = os.path.abspath(os.path.join(os.path.dirname(__file__), "data"))
+        standardpath = os.path.join(self.datadir, "standard")
+        externalpath = os.path.join(self.datadir, "external")
+        self.queue = scriptqueue.ScriptQueue(index=1,
+                                             standardpath=standardpath,
+                                             externalpath=externalpath)
+        self.queue.summary_state = salobj.State.DISABLED
+        self.queue_remote = salobj.Remote(SALPY_ScriptQueue, index=1)
+        self.process = None
+
+        self.scheduler = SchedulerCSC(index=1)
+        self.scheduler.parameters.mode = 'SIMPLE'
+        self.scheduler_remote = salobj.Remote(SALPY_Scheduler, index=1)
+        self.received_targets = 0
+        self.expected_targets = 2
+
+        self.heartbeats = 0
+        self.heartbeats_tol = 0.4
+
+        self.target_test_timeout = 120
+
+    async def enable_scheduler(self):
+        """ Utility method to enable the scheduler CSC. """
+
+        # send enterControl; new state is STANDBY
+        state_coro = self.scheduler_remote.evt_summaryState.next(timeout=1.)
+        cmd_attr = getattr(self.scheduler_remote, f"cmd_enterControl")
+        await cmd_attr.start(cmd_attr.DataType(), timeout=1.)
+        state = await state_coro
+        self.assertEqual(state.summaryState, salobj.State.STANDBY,
+                         "Scheduler in %s, expected STANDBY. " % salobj.State(state.summaryState))
```

Comment:
`next(timeout=1.)` will not work with current ts_salobj: the `flush` argument always has to be specified now. But instead of fixing that, I suggest a bigger change:

Replace `state_coro = self.scheduler_remote.evt_summaryState.next` with:
```
state = await self.scheduler_remote.evt_summaryState.next(flush=False, timeout=1)
self.assertEqual(state.summaryState, ...)
```
and check that it is STANDBY. Then after each `cmd_<name>.start(...)` call use that same exact pair of lines. The result is much simpler code that is easier to read. (I have updated ts_salobj to work the same way).

Anywhere you have `state_coro = await ...` you can do the same.